src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -167,18 +178,26 @@ const Badges: React.FC<{ user: WithId<User> }> = ({ user }) => {
     return playaHours > 1 ? `${playaHours}` : "< 1";
   }, [visits]);
 
+  // Only show visits to related venues
   const venuesVisited = useMemo(() => {
     if (!visits) return undefined;
-    return visits.filter((visit) => visit.id !== PLAYA_VENUE_NAME).length; // Playa does not count
-  }, [visits]);
+    return visits.filter(
+      (visit) =>
+        currentVenue.id === visit.id || relatedVenueIds.includes(visit.id)
+    ).length;
+  }, [visits, currentVenue, relatedVenueIds]);
 
   const badges = useMemo(() => {
     if (!visits || !venues) return [];
     return visits
       .filter((visit) => visit.id !== PLAYA_VENUE_NAME) // no badge for the Playa. Also does not have a logo
       .map((visit) => {
         const { venue, room } = findVenueAndRoomByName(visit.id, venues);
-        if (!venue) return undefined;
+        if (
+          !venue ||
+          (currentVenue.id !== venue.id && !relatedVenueIds.includes(venue.id))
+        )
+          return undefined;
```

Comment:
Are these basically doing the same processing? Except the first one just gets the length of it, whereas the second one maps them to some things?

I'm wondering if we can process the array once, then read the count straight off that.

I'm not deeply following the logic here, so unsure if that is possible.

Reply:
Yes, they can be deduped, just did! Good call.